app/src/main/java/com/zulip/android/activities/RecyclerMessageAdapter.java

```diff
@@ -353,17 +370,41 @@ public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
 
                 MessageHolder messageHolder = ((MessageHolder) holder);
                 final Message message = ((Message) items.get(position));
+
                 messageHolder.contentView.setText(message.getFormattedContent(zulipApp));
 
+                final String url = message.extractImageUrl(zulipApp);
```

Comment:
This is a "fragile" approach as of now. If we proceed in this direction we probably should have a different way of retrieving urls.